transaction/core/src/input_rules.rs

```diff
@@ -53,6 +69,116 @@ impl InputRules {
                 return Err(InputRuleError::MissingRequiredOutput);
             }
         }
+
+        self.verify_partial_fill_rules(block_version, tx)?;
+
+        Ok(())
+    }
+
+    // Partial-Fill rules verification (MCIP #42)
+    fn verify_partial_fill_rules(
+        &self,
+        _block_version: BlockVersion,
+        tx: &Tx,
+    ) -> Result<(), InputRuleError> {
+        if let Some(partial_fill_change) = self.partial_fill_change.as_ref() {
+            // There is a partial fill change output. Let's try to unblind its amount.
+            let (partial_fill_change_amount, _) = partial_fill_change.reveal_amount()?;
+            // If the min fill value exceeds the fractional change, then the SCI is
+            // ill-formed
+            if self.min_partial_fill_value > partial_fill_change_amount.value {
```

Comment:
nit: I think it's slightly easier to understand that the `partial_fill_change_amount.value < min_partial_fill_value` and would recommend rewriting as such. Just my opinion however.